snuba/clickhouse/http.py

```diff
@@ -199,7 +200,7 @@ def append(self, value: bytes) -> None:
         assert not self.__closed
 
         self.__queue.put(value)
-        self.__rows += 1
+        self.__rows.append(value)
```

Comment:
How big (in length and in bytes) is this expected to get?

Reply:
I don't know to be honest. We have many different datasets with different column counts and batch sizes. I might want to deploy a PR first that estimates this via DD metric but perhaps somebody can make an assessment straight away on whether this approach is feasible or whether I need to get more clever.

Reply:
This DD notebook might be a rough estimate of row sizes/dataset based on total bytes inserted vs. total row count inserted:

https://app.datadoghq.com/notebook/3950488/oliver-nov-17-2022-13-23

Reply:
Why not as an initial step, you start emitting metrics for `self.__size` to get a sense of how big the total length of value is? Events and Transactions payload can be really big. And the processing strategy which writes to clickhouse happens in parallel to batches being filled with more messages from kafka concurrently. So in theory the memory usage might be close to doubling with this.

Reply:
This is a good idea nikhar, done in https://github.com/getsentry/snuba/pull/3431